jujugui/static/gui/src/app/app.js

```diff
@@ -1588,7 +1588,80 @@ YUI.add('juju-gui', function(Y) {
       @method _handleNewModel
     */
     _handleNewModel: function() {
-      this.switchEnv(null, null, null, null, false, true);
+      this.switchEnv();
+    },
+
+    /**
+      Handle the request to display the user entity state.
+
+      @method _handleUserEntity
+      @param {Object} state - The application state.
+      @param {Function} next - Run the next route handler, if any.
+      @param {Boolean} lookupModels - Whether to fetch models.
+    */
+    _handleUserEntity: function(state, next, lookupModels=true) {
+      if (!this.userPaths) {
+        this.userPaths = {};
+      }
+      const userPath = state.user;
+      const match = this.userPaths[userPath];
+      if (match) {
+        if (match.type === 'store') {
+          this._renderCharmbrowser(state, next);
+        } else if (match.type === 'model') {
+          const uuid = match.model.uuid;
```

Comment:
grab the code in this block and put it into it's own nested function within `_handleUserEntity`, assuming we call it `connectToModel`